app/components/model-renderer.jsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import Model from './modelling';
+
+// function to check whether webgl is available
+function webGLCompatibilityTest() {
+  try {
+    return !!window.WebGLRenderingContext && !!document.createElement('canvas').getContext('experimental-webgl');
+  } catch (e) {
+    return false;
+  }
+}
+
+class ModelRenderer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.getDifference = this.getDifference.bind(this);
+    this.setTexture = this.setTexture.bind(this);
+
+    this.imgUrl = Object.values(this.props.subject.locations[0])[0];
+    if (this.props.subject.locations.length < 2) {
+      const im0 = this.props.subject.locations[0];
+      this.props.subject.locations[1] = Object.assign({}, im0);
+    }
+  }
+  componentDidMount() {
+    if (!webGLCompatibilityTest()) {
+      // TODO: is there a fallback option?
+      /* eslint-disable no-console */
+      console.error('WebGL is not available, aborting');
```

Comment:
It might be useful to render a message to the so the user realizes that the project will not work in their browser.  Something like: "It looks like your browser will not work for this project, please use XXX or YYY"